compiler/passes/checkParsed.cpp

```diff
@@ -314,6 +317,21 @@ checkFunction(FnSymbol* fn) {
   }
 }
 
+static void nestedName(ModuleSymbol* mod) {
+  if (mod->defPoint == NULL) {
+    return;
+  }
+
+  ModuleSymbol* parent = mod->defPoint->getModule();
+  if (parent != mod &&
+      strcmp(mod->name, parent->name) == 0 &&
```

Comment:
I'm pretty sure you should be able to pointer-compare the names here because of 'astr'